src/amplitude-snippet.js

```diff
@@ -7,25 +7,30 @@
   as.onload = function() {window.amplitude.runQueuedFunctions();};
   var s = document.getElementsByTagName('script')[0];
   s.parentNode.insertBefore(as, s);
-  var Identify = function() {this._q = []; return this;};
-  function proxyIdentify(fn) {
-    Identify.prototype[fn] = function() {
+  function proxy(obj, fn) {
+    obj.prototype[fn] = function() {
       this._q.push([fn].concat(Array.prototype.slice.call(arguments, 0))); return this;
     };
   }
+  var Identify = function() {this._q = []; return this;};
   var identifyFuncs = ['add', 'append', 'clearAll', 'prepend', 'set', 'setOnce', 'unset'];
-  for (var i = 0; i < identifyFuncs.length; i++) {proxyIdentify(identifyFuncs[i]);}
+  for (var i = 0; i < identifyFuncs.length; i++) {proxy(Identify,identifyFuncs[i]);}
   amplitude.Identify = Identify;
+  var Revenue = function() {this._q = []; return this;};
```

Comment:
We have a stand-in "proxy" revenue object that intercepts any calls to new amplitude.Revenue(). It saves function calls and their arguments to an internal queue _q. Once the async loading of our script finishes, then we can convert the proxy object into the real object. That happens in logRevenueV2, we iterate through _q and apply the functions to an actual instance of Revenue.

We already do this for Identify objects, and for the main amplitude object